hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -313,4 +345,114 @@ public String getQuorumName(final String name) {
         return quorumName == NULL_OBJECT ? null : (String) quorumName;
     }
 
+    private MergePolicyConfig getMergePolicyConfig(String name) {
+        return getNodeEngine().getConfig().getScheduledExecutorConfig(name)
+                              .getMergePolicyConfig();
+    }
+
+    private SplitBrainMergePolicy getMergePolicy(String name) {
+        return mergePolicyProvider.getMergePolicy(
+                getMergePolicyConfig(name).getPolicy());
+    }
+
+    private class Merger implements Runnable {
+
+        private static final int TIMEOUT_FACTOR = 500;
+
+        private Map<Integer, Map<String, Collection<ScheduledTaskDescriptor>>> partitionsSnapshot;
+
+        Merger(Map<Integer, Map<String, Collection<ScheduledTaskDescriptor>>> map) {
+            this.partitionsSnapshot = map;
+        }
+
+        @SuppressWarnings({"checkstyle:methodlength"})
+        @Override
+        public void run() {
+            // we cannot merge into a 3.9 cluster, since not all members may understand the CollectionMergeOperation
```

Comment:
There is no `CollectionMergeOperation` in this data structure ;) But anyway we have now logging in case of a 3.9 cluster, so the block is like this in all other services:
```java
            final ILogger logger = nodeEngine.getLogger(DistributedScheduledExecutorService.class);
            final Semaphore semaphore = new Semaphore(0);

            ExecutionCallback<Object> mergeCallback = new ExecutionCallback<Object>() {
                @Override
                public void onResponse(Object response) {
                    semaphore.release(1);
                }

                @Override
                public void onFailure(Throwable t) {
                    logger.warning("Error while running merge operation: " + t.getMessage());
                    semaphore.release(1);
                }
            };

            // we cannot merge into a 3.9 cluster, since not all members may understand the MergeOperation
            if (nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V3_10)) {
                logger.info("Cluster needs to run version " + Versions.V3_10 + " to merge scheduled exector instances");
                return;
            }
```